tests/serverpod_test_server/test_e2e/auth_core_test.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:test/expect.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('Given a simulated legacy client with old authorization conventions, ',
+      () {
+    late Client client;
+    late String authKey;
+
+    setUpAll(() async {
+      // prepare a proper authentication key
+      client = Client(
+        serverUrl,
+        authenticationKeyManager: TestAuthKeyManager(),
+      );
+      var response =
+          await client.authentication.authenticate('test@foo.bar', 'password');
+      assert(response.success, 'Authentication setup failed');
+      authKey = '${response.keyId}:${response.key}';
+    });
+
+    tearDownAll(() async {
+      client.close();
+    });
```

Comment:
Suggestion: Add removal of the authenticated user to the teardown.

We should try and clean up any artifact created from the test. You can view such a cleanup here: https://github.com/serverpod/serverpod/blob/a3f8ab507b858146e30f0eac207249c12eb3eb59/tests/serverpod_test_server/test_e2e/method_streaming/authenticated_streaming_method_test.dart#L49-L57


I'm not sure that the assert at the end is required.